packages/xstate-wallet/src/store/memory-store.ts

```diff
@@ -67,16 +68,24 @@ interface InternalEvents {
   channelUpdated: [ChannelStoreEntry];
   newObjective: [Objective];
   addToOutbox: [Message];
+  ledgerUpdated: [LedgerUpdated];
 }
 
+export type LedgerStatus = {
+  ledgerId: string;
+  lock?: Guid;
+};
+export type LedgerUpdated = LedgerStatus & {participantId: string};
+type LedgerLocked = LedgerUpdated & {lock: Guid};
+
 export class MemoryStore implements Store {
   readonly chain: Chain;
   protected _channels: Record<string, MemoryChannelStoreEntry | undefined> = {};
   private _objectives: Objective[] = [];
   private _nonces: Record<string, BigNumber | undefined> = {};
   private _eventEmitter = new EventEmitter<InternalEvents>();
   private _privateKeys: Record<string, string | undefined> = {};
-  protected _ledgers: Record<string, string | undefined> = {};
+  protected _ledgers: Record<string, LedgerStatus | undefined> = {};
```

Comment:
Perhaps name this more explicitly as `_ledgerChannelLockStatuses`?

Reply:
It's also the lookup for the `ledgerId` for a given `participantId`

Reply:
Is that necessary to cache assuming it is inside the `MemoryChannelStoreEntry`?

Reply:
I don't quite follow.

On master, `_ledgers` serves this purpose:
1. I want to virtually fund `C` through `H`
2. I get to the step where I need to indirectly fund my guarantor
3. I need to find the `ledgerId` with the hub

The `ledgerId` is not stored on any `channelStoreEntry` until it is determined, and `_ledgers` is the discovery mechanism.

I think `_ledgers` should probably remain unchanged, and I can just add a `_locks: Record<channelId, Guid>` lookup.